modules/core/src/opencl/runtime/opencl_clblast.cpp

```diff
@@ -0,0 +1,92 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../../precomp.hpp"
+
+#ifdef HAVE_CLBLAST
+
+#include "opencv2/core/opencl/runtime/opencl_core.hpp"
+#include "opencv2/core/opencl/runtime/opencl_clblast.hpp"
+
+#if defined(_WIN32)
+#include <windows.h>
+
+    static void* WinGetProcAddress(const char* name) {
+        static HMODULE opencl_module = NULL;
+        if (!opencl_module) {
+            opencl_module = GetModuleHandleA("clblast.dll");
+            if (!opencl_module) {
+                opencl_module = LoadLibraryA("clblast.dll");
+                if (!opencl_module)
+                    return NULL;
+            }
+        }
+        return (void*)GetProcAddress(opencl_module, name);
+    }
+    #define CV_CL_GET_PROC_ADDRESS(name) WinGetProcAddress(name)
+#endif // _WIN32
+
+#if defined(__linux__) || defined(__APPLE__)
+    #include <dlfcn.h>
+    #include <stdio.h>
+
+    static void* GetProcAddress (const char* name) {
+        static void* h = NULL;
+        if (!h) {
+            #if defined (__linux__)
+            h = dlopen("libclblast.so", RTLD_LAZY | RTLD_GLOBAL);
+            #endif
+            #if defined (__APPLE__)
+            h = dlopen("libclblast.dylib", RTLD_LAZY | RTLD_GLOBAL);
+            #endif
+            if (!h) {
+                printf("handle is null\n");
```

Comment:
more clear message is preferable. The current form of message doesn't help to user to resolve problem.

Reply:
Done